test/tests/functional/pbs_snapshot_unittest.py

```diff
@@ -0,0 +1,208 @@
+        # Check 4: qstat_Bf.out has 'job_history_duration' set to 24:00:00
+        with open(qstat_bf_path, "r") as fd:
+            for line in fd:
+                if "job_history_duration" in line:
+                    # Remove whitespaces
+                    line = "".join(line.split())
+                    # Split it up by '='
+                    key_val = line.split("=")
+                    self.assertTrue(key_val[1] == job_hist_duration)
+
+        # Cleanup
+        self.du.rm(path=outdir_fullpath, recursive=True, force=True)
+
+    def test_capture_all(self):
+        """
+        Test the 'capture_all' interface of PBSSnapUtils
+        Note: this will test out all capture_ interfaces of PBSSnapUtils
+        except capture_pbs_logs
+        """
+
+        output_dir = "snapshot_all"
+        snap_args = {"out_dir": output_dir, "num_service_logs": 2,
+                     "num_acct_logs": 5}
+
+        with PBSSnapUtils(snap_args, create_tar=True) as snap_obj:
+            snap_obj.capture_all()
+
+        # Cleanup
+        self.du.rm(path=output_dir, recursive=True, force=True)
+
+    def test_capture_pbs_logs(self):
+        """
+        Test the 'capture_pbs_logs' interface of PBSSnapUtils
+
+        WARNING: Assumes that the test is being run on type - 1 PBS install
+        """
+        output_dir = "snapshot_all"
+        snap_args = {"out_dir": output_dir, "num_service_logs": 2,
+                     "num_acct_logs": 5}
+
+        with PBSSnapUtils(snap_args) as snap_obj:
+            snap_obj.capture_pbs_logs()
+
+        # Perform some checks
+        # Check that the snapshot exists
+        self.assertTrue(os.path.isdir(output_dir))
+        # Check that 'server_logs' were captured
+        log_path = os.path.join(output_dir, "server", "server_logs")
+        self.assertTrue(os.path.isdir(log_path))
+        # Check that 'mom_logs' were captured
+        log_path = os.path.join(output_dir, "node", "mom_logs")
+        self.assertTrue(os.path.isdir(log_path))
+        # Check that 'comm_logs' were captured
+        log_path = os.path.join(output_dir, "comm", "comm_logs")
+        self.assertTrue(os.path.isdir(log_path))
+        # Check that 'sched_logs' were captured
+        log_path = os.path.join(output_dir, "scheduler", "sched_logs")
+        self.assertTrue(os.path.isdir(log_path))
+        # Check that 'accounting_logs' were captured
+        log_path = os.path.join(
+            output_dir, "server", "server_priv", "accounting")
+        self.assertTrue(os.path.isdir(log_path))
+
+        # Cleanup
+        self.du.rm(path=output_dir, recursive=True, force=True)
+
+    def test_snapshot_basic(self):
+        """
+        Test capturing a snapshot via the pbs_snapshot program
+        """
+        output_dir = "mysnapshot_basic"
+        snap_cmd = ["pbs_snapshot", "-o", output_dir]
+        ret = self.du.run_cmd(cmd=snap_cmd)
+        self.assertTrue(ret['rc'] == 0)
```

Comment:
Can you test that you actually did a snapshot instead of just checking for a 0 return code?